src/olympia/amo/utils.py

```diff
@@ -986,7 +985,13 @@ def utc_millesecs_from_epoch(for_datetime=None):
     """
     if not for_datetime:
         for_datetime = datetime.datetime.now()
-    return calendar.timegm(for_datetime.utctimetuple()) * 1000
+    # Number of seconds.
+    seconds = time.mktime(for_datetime.utctimetuple())
+    # timetuple() doesn't care about more precision than seconds, but we do.
+    # Add microseconds as a fraction of a second to keep the precision.
+    seconds += for_datetime.microsecond / 1000000.
```

Comment:
Are you _sure_ this works as expected?  I checked in a shell and this expression always resolves to an integer value, meaning we instantly lose the fraction of a second you just added.  You'd need `float(...)` around the microsecond value, afaik.  Or have `seconds` be `milliseconds` on the line above and just divide `.microsecond` by `1000` instead

Reply:
`microsecond` is converted into a float by dividing with a float `1000000` has the dot at the end.

Reply:
```python
>>> seconds = time.mktime(for_datetime.utctimetuple())
>>> seconds
1542711112.0
>>> seconds += for_datetime.microsecond / 1000000.
>>> seconds
1542711112.344284
```

Everything looks ok ?

Reply:
doh, I missed the `.`

Reply:
`1000000.0` for skim readers like myself?